Update Docs Release Timeline after the 1.21 release (#1410)
kubernetes/sig-release: release-team/role-handbooks/docs/Release-Timeline.md

    - [kubernetes-release-team-shadows](https://groups.google.com/a/kubernetes.io/g/release-team-shadows)
        - ⚠️  List of members for this group is managed in git. Please create a PR against `kuberenetes/sig-release` repo to include your email under `release-team-shadows` group in [`sig-release/group.yaml`](https://github.com/kubernetes/k8s.io/blob/master/groups/sig-release/groups.yaml)

- Make sure you're included in the "official" release team file, e.g: [release 1.21](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.21/release-team.md). If not submit a PR and add yourself.

> Early in the release cycle, the Release Manager opens an enhancement tracking spreadsheet, e.g: [the 1.21 release spreadsheet](https://docs.google.com/spreadsheets/d/1I_ybuEI6gNNPJT7xcO_y-eu_BpayTdBX3nerVsHybNA/edit?usp=sharing). Later in the release this spreadsheet will contain important information for docs:

Reviewer: ```suggestion
Early in the release cycle, the Enhancements Lead opens an enhancement tracking spreadsheet, e.g: [the 1.21 release spreadsheet](http://bit.ly/k8s121-enhancements). Later in the release this spreadsheet will contain important information for docs:
```

- Which enhancements we'll be tracking for this release
- Feature owners (and their GitHub IDs)
- Links to docs PRs opened for each enhancement

### Read the release timeline

Read the release timeline and **make sure the timeline includes deadlines for documentation work**, e.g: [1.21 timeline](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.21#timeline):

- Docs deadline - Open placeholder PRs (~3.5 weeks before release)
- Docs deadline - PRs ready for review (~2 weeks before release)
- Docs complete - All PRs reviewed and ready to merge (~1 week before release)

If these deadlines aren't listed in the release timeline, request that the Release Lead add them.
Define deadlines clearly e.g. eod PST is not clear if eod is 5pm or 11:59am PST.

### Introduce yourself to docs localization teams

> Create a GitHub discussion in [k/sig-release](https://github.com/kubernetes/sig-release/discussions/ ) to introduce yourself to the current localization owners and to sync up early on the release timeline, e.g:

Reviewer: ```suggestion
Create a GitHub discussion in [k/sig-release](https://github.com/kubernetes/sig-release/discussions/) to introduce yourself to the current localization owners and to sync up early on the release timeline, e.g:
```

> Title: K8s 1.22 - Formalize docs release strategy with docs localization owners
>
> Hello :wave:,
> I'm < docs lead name >, the SIG Release Docs lead for the 1.22 release. I'm opening this discussion to coordinate the docs release (scheduled tentnatively for [< date >](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.22#timeline)) with localization owners.
> I look forward to working with everyone! Let me know if you have any questions!
> Thank you! 🙏
>
> cc: @kubernetes/sig-docs-fr-owners @kubernetes/sig-docs-de-owners @kubernetes/sig-docs-hi-owners @kubernetes/sig-docs-id-owners @kubernetes/sig-docs-ja-owners @kubernetes/sig-docs-ru-owners @kubernetes/sig-docs-pl-owners @kubernetes/sig-docs-ko-owners @kubernetes/sig-docs-pt-owners @kubernetes/sig-docs-es-owners @kubernetes/sig-docs-zh-owners @kubernetes/sig-docs-vi-owners @kubernetes/sig-docs-uk-owners @kubernetes/sig-docs-it-owners

> - Try to avoid tagging all localization owners using the `@kubernetes/sig-docs-xx-owners` until it becomes necessary.

Reviewer: Can we remove this line if we're no longer collecting one individual contact and tagging all localization owners in the discussion?

### Ensure access is set up

1. Add shadows to the "official" release team file, e.g.: [1.21 release](https://github.com/kubernetes/sig-release/pull/1432)

1. As a lead, make sure you are part of the [sig-docs-en-owners](https://github.com/orgs/kubernetes/teams/sig-docs-en-owners), e.g: [1.21 release](https://github.com/kubernetes/website/pull/26114).

1. Add docs team to the `website-milestone-maintainers` group in k/org/config/kubernetes/sig-docs/teams.yaml and `release-team` group in k/org/kubernetes/sig-release/teams.yaml, e.g: [1.21 release](https://github.com/kubernetes/org/pull/2445).

1. Make sure all shadows have edit access to the enhancement spreadsheet. Contact the enhancements lead to grant access for you and shadows.

1. You need push access to the Kubernetes website repo (contact a SIG Docs chair if you don't have it)

>    ⚠️ (**This should be done by the Docs Lead**) Open the [integration branch] by creating a pull request through the GitHub UI against `master` referencing the `dev-[future release]` branch e.g., the [Release 1.14](https://github.com/kubernetes/website/pull/13174) PR uses the branch `dev-1.14`.

Reviewer 2: since the examples above point to v1.21, would it make sense to update all of them to match, instead of point to different examples from different releases?

Reviewer: I don't think that matters too much since the way the PR should be made has not changed. If the process has changed, I believe the example PR also changed with it to ensure that latest examples are listed in the doc.

Author: Updated examples to 1.21

   This release pull request (also known as the [integration branch]) serves as the base for individual, component enhancement PRs of the release. A [integration branch] lets you bundle and merge multiple PRs simultaneously.

   ⚠️ Add the label `do-not-merge/hold` to the PR.

> 1. First PR in `dev-[future release]`: Update config.toml to show `[future release]` as the current version and add the `[future release]` entry to the drop-down, e.g: [config.toml diff](https://github.com/kubernetes/website/pull/20847).

Reviewer 2: regarding my version comment this points to some PR for v1.19

> Use the [patch releases](https://github.com/kubernetes/sig-release/blob/master/releases/patch-releases.md) as a reference.

Reviewer: Modifying the previous patch versions are not required for the first `config.toml` change. Not sure if it's useful here since it's referenced in L539

This is very helpful for weekly reports and managing deadlines

### Communicate major deadlines

COMMUNICATE ALL 3 MAJOR DATES AT LEAST A WEEK PRIOR, INCLUDING THE RELEASE DATE REPO FREEZE, FOLLOWING THE BELOW METHODS:

- Keep developers apprised of due dates for docs.
    Send announcements to:

>     - [Kubernetes Dev Group](https://groups.google.com/forum/#!forum/kubernetes-dev)

Reviewer: Is the announcement to k-dev necessary for Docs Lead? I think both Enhancements Lead and the Release Lead takes care of most of the announcements related to the release for k-dev. We should promote communication between the subteams and make sure important dates are communicated without needing Docs Lead to send out a separate announcement.

Author: I agree that the Enhancements Lead and the Release Lead take care of announcements to k-dev. I'll modify this list

    - [#sig-release](https://kubernetes.slack.com/messages/sig-release) on Slack
    - Your team / shadows
    - [#sig-docs](https://kubernetes.slack.com/messages/sig-docs)

>     - [#sig-docs-maintainers]((https://kubernetes.slack.com/messages/sig-docs-maintainers))

Reviewer: ```suggestion
    - [#sig-docs-maintainers](https://kubernetes.slack.com/messages/sig-docs-maintainers)
```

If no suitable candidates are available, you may choose to lead again or nominate someone outside of the current release team (e.g: a recent Docs Lead).

> ## Late-Middle Steps (Weeks 7-8) - Release Blog Progress and SIG Docs (~4 weeks before the release)

Reviewer: There is a section here about reaching out to comms: https://github.com/kubernetes/sig-release/blob/master/release-team/role-handbooks/docs/Release-Timeline.md#touch-base-with-the-release-communications-team

Can we remove that as part of this PR and keep this section?

> Reach out to the Communications and Enhancements Teams about the progress on the Release Blog.

Reviewer 3: wondering if this needs to be more specific? bc i can't really tell what this is^^ are you expecting feedback from those teams? or just telling them that you started? what is the purporse is my question i guess.

Author: @kikisdeliveryservice I put the PR on hold for now. This will be flushed out as we define how Docs should work with Comms, Enhancements, and Release Notes e.g. any of the teams can relay to Release Notes about any enhancements or deprecations that will be good for the Major Themes section of the release notes

* 1.17 https://github.com/kubernetes/website/pull/25391
* 1.18 https://github.com/kubernetes/website/pull/25390
* 1.19 https://github.com/kubernetes/website/pull/25467

Changes required:
- Set `latest` to the [future release] version
- Update `fullversion` and `githubbranch` to the latest patched version of the corresponding release
- Update the list of versions to include the [future release] and remove the oldest release
- Set `deprecated` to `true`
- For each previous release, update the patch version to the latest patch version available

⚠️  DO NOT MERGE **ANY** OF THE CONFIGURATION PULL REQUESTS UNTIL THE RELEASE HAS OCCURRED

### Update the site configuration files for future release

The future release branch site configuration also needs to be updated to reflect the latest patch versions of the past releases.
Create a PR against the `dev-[future-release]` branch to update the `config.toml` file that matches the
[PRs for previous releases](#update-the-site-configuration-files-for-previous-releases).

### Inform localization teams

Let localization team know about freeze and next tentative timeline(s) for important dates

> > Hello localization team leads! We talked about docs in v1.14 here (https://github.com/kubernetes/website/issues/12396). I don't think any action is required from you, but I wanted to let you know that we are on track for the release (3/25/19) and all Kubernetes website branches are up to date (master, dev-1.14, release-1.13). Let me know if I can help with anything! Thanks!

Reviewer: ```suggestion
> Hello localization team leads! I don't think any action is required from you, but I wanted to let you know that we are on track for the release (3/25/19) and all Kubernetes website branches are up to date (master, dev-1.14, release-1.13). Let me know if I can help with anything! Thanks!
```

851ef58fa - Jim Angel, 5 hours ago : Official 1.14 Release Docs (#13174)
28dd4d515 - evilyeti, 7 hours ago : Fixed broken link in E2E Testing blog
```
If that looks how you expect, the following example is from the 1.14 release:
```shell
git tag -a snapshot-final-v1.13 28dd4d515 -m "Release 1.13 final snapshot"
git tag -a snapshot-initial-v1.14 851ef58fa -m "Release 1.14 initial snapshot"
git push --tags origin master
```

> If the `git log` command does not look how you expect then you can use the following to confirm the commit hash before the [integration branch]:

Reviewer: Can we just get rid of L713-726, and use the most recent example? The example log causes more confusion and I think it would be better to remove it for clarity.

```shell
git show <merge commit of integration branch>^1
```
The following example is from the 1.21 release when the `git log` command did not output the expected commit hashes.
It was noted that the last commit hash of the `release-1.20` branch is 6d252624b21452bcf6c22f156cccd316111204d4 from PR 2738 and the commit hash of the integration branch merge which is the first commit of 1.21 is 969a3db92326746a94e2eb4449fd60f25d9151b8.
```shell 
git show 969a3db92326746a94e2eb4449fd60f25d9151b8^1

commit 6d252624b21452bcf6c22f156cccd316111204d4 (origin/release-1.20)
Merge: d1c650d2a 5767f39fd
Author: Kubernetes Prow Robot <k8s-ci-robot@users.noreply.github.com>
Date:   Wed Apr 7 07:53:53 2021 -0700

    Merge pull request #27387 from Arhell/label
    
    [ja] updated labels for cli command in the PHP Guestbook tutorial
```
Proceed with tagging the commit hashes:
```shell
git tag -a snapshot-final-v1.20 6d252624b -m "Release 1.20 final snapshot"

> git tag -a snapshot-initial-v1.21 969a3db92 -m "Release 1.21 initial snapshot"

Reviewer: ```suggestion
git tag -a snapshot-initial-v1.21 969a3db92 -m "Release 1.21 initial snapshot"
git push --tags origin master
```
